### PR TITLE
Upgrade to webpack 5

### LINKED ISSUE #31: Upgrade to webpack 5
We need to keep our dependencies to latest versions.
The biggest upgrade is related to webpack.
Moving to webpack 5 will also improve final bundles performance and reduce amount of webpack specific code comparing to v4.
https://webpack.js.org/migrate/5/

### PR DESCRIPTION
**Pull request by juantxokupul**
## Description

Upgrade to webpack 5

## Related Issue

Fixes #31 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **[CONTRIBUTING](docs/CONTRIBUTING.mnd)** document.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by raohmaru** [comment]
Please wait until PR #75 is merged, it contains new test cases to test the build.

**Comment by easingthemes** [comment]
This looks to easy to be truth :). 
We have delayed upgrade for almost a year because we expected extremely big changes.

1. It should be a breaking change, since you can't know how users extended this library. Eg we are using some webpack/babel plugins by extending `.febuild` and we need to update (or even totally replace) them since they are not compatible with webpack v5.
2. It should be tested, as @raohmaru mentioned

**Review by easingthemes — CHANGES_REQUESTED** [review]
It should be a breaking change, since you can't know how users extended this library. Eg we are using some webpack/babel plugins by extending .febuild and we need to update (or even totally replace) them since they are not compatible with webpack v5.
It should be tested, as @raohmaru mentioned

**Comment by juantxokupul** [comment]
Leaving this unassigned since currently I don't find time to work on this.

**Review by quicoto — APPROVED** [review]

**Comment by raohmaru** [comment]
Does it contain any breaking change? If so it should be detailed in the readme.

**Comment by quicoto** [comment]
> Leaving this unassigned since currently I don't find time to work on this.

I think so, we need to push a commit with "BREAKING CHANGE: " so the build will trigger a major release.

File: utils/__snapshots__/linterError.test.js.snap
Comment (easingthemes) [inline utils/__snapshots__/linterError.test.js.snap:6]: This path looks wrong?

**Review by easingthemes — APPROVED** [review]

File: utils/__snapshots__/linterError.test.js.snap
Comment (FelipeSimoes) [reply utils/__snapshots__/linterError.test.js.snap:6]: It's correct it's ```const source = __filename;``` at linterError.test.js

**Pull request merged**
